### ISSUE TITLE
[FEATURE]: chazel-atom

### ISSUE THREAD

**Issue by dpiliouras** [tolitius/chazel#10]
Hi there, 

I'm sure you're aware but one of Hazelcast's offerings is a distributed `IAtomicReference`. On top of this, one can build a distributed atom - e.g. https://github.com/jimpil/hazel-atom/blob/master/src/hazel_atom/core.clj. 

Are you interested in this? Perhaps the implementation can be further improved given Hazelcast v5, or maybe not - i haven't exactly looked at it recently...in any case, having a distributed atom would be a neat feature, right?

**Comment by tolitius** [comment]
yes, I [see](https://docs.hazelcast.com/hazelcast/5.1/data-structures/iatomicreference) it in docs.

what would be the use case: i.e. why/when would you need to use it?
the doc says:

> _but in some cases you need to deal with a [distributed] reference_...
> _`IAtomicReference` works based on the byte-content and not on the object-reference. If you use the `compareAndSet` method, do not change to the original value because its serialized content will then be different_

curious what would be a practical usage for this
